src/codegen/compile.ml

```diff
@@ -2706,17 +2708,18 @@ module Object = struct
 
 
   (* Returns a pointer to the object field (without following the indirection) *)
-  let idx_hash_raw env =
-    Func.share_code2 env "obj_idx" (("x", I32Type), ("hash", I32Type)) [I32Type] (fun env get_x get_hash ->
+  let idx_hash_raw env low_bound =
+    let name = Printf.sprintf "obj_idx<%d>" low_bound  in
+    Func.share_code2 env name (("x", I32Type), ("hash", I32Type)) [I32Type] (fun env get_x get_hash ->
       let (set_h_ptr, get_h_ptr) = new_local env "h_ptr" in
 
       get_x ^^ Heap.load_field hash_ptr_field ^^ set_h_ptr ^^
 
       get_x ^^ Heap.load_field size_field ^^
       (* Linearly scan through the fields (binary search can come later) *)
-      from_0_to_n env (fun get_i ->
+      from_m_to_n env (Int32.of_int low_bound) (fun get_i ->
         get_i ^^
-        compile_mul_const Heap.word_size  ^^
+        compile_mul_const Heap.word_size ^^
```

Comment:
Hmm, another comment seems to have been lost by GH.
I think I suggested turning the for loop into a while loop, using pointer addition instead of multplication on each round.
Something like

let cur = htp_ptr + low_bound * n
while (*hp_ptr <> hash) {
   hp_ptr += 4
};
return hp_ptr

Reply:
Yes, I have seen this. Strength reduction would be nice here. Also there is really no need for the end condition, since the hash is necessarily present if our type system is sound :-)

Reply:
Again, that could be left as an optimization to a separate PR.

Reply:
Oh, just see you say the same thing :-)

Reply:
Probably only worth considering when the measurements are back in place (and totally fine as a separate PR).

Reply:
This is merged as #2722.